packages/slice-machine/src/features/environments/actions/getActiveEnvironment.ts

```diff
@@ -0,0 +1,34 @@
+import {
+  Environment,
+  isInvalidActiveEnvironmentError,
+} from "@slicemachine/manager/client";
+
+import { managerClient } from "@src/managerClient";
+
+type GetActiveEnvironmentReturnType =
+  | {
+      activeEnvironment: Environment;
+      error?: undefined;
+    }
+  | {
+      activeEnvironment?: undefined;
+      error: unknown;
+    };
+
+export async function getActiveEnvironment(): Promise<GetActiveEnvironmentReturnType> {
+  try {
+    const { activeEnvironment } =
+      await managerClient.project.fetchActiveEnvironment();
+
+    return { activeEnvironment };
+  } catch (error) {
+    if (isInvalidActiveEnvironmentError(error)) {
+      // Reset to the production environment.
+      await managerClient.project.updateEnvironment({ environment: undefined });
+
+      return await getActiveEnvironment();
```

Comment:
Can this create an infinite loop here?

Reply:
It could if `updateEnvironment({ environment: undefined })` continued to set an invalid environment. That should only happen if the adapter does not adhere to the convention that `undefined` is the production environment.

Do you have a suggestion on handling it? We could attempt changing to the production environment once and fail hard if it does not change.

I'm not sure how we handle the failure case, however. It theoretically shouldn't happen, so a full-app error message may be okay.

Reply:
For me, we should have only one fallback to "Production", yes I guess.